target_chains/sui/contracts/sources/pyth.move

```diff
@@ -85,8 +92,50 @@ module pyth::pyth {
         sources
     }
 
-    /// Create and share new price feed objects if they don't already exist.
-    public fun create_price_feeds(
+    /// Create and share new price feed objects if they don't already exist using accumulator message.
+    public fun create_price_feeds_using_accumulator(
+        pyth_state: &mut PythState,
+        accumulator_message: vector<u8>,
+        vaa: VAA, // the verified version of the vaa bytes encoded within the accumulator_message
```

Comment:
I suggest making a struct to encapsulate these two arguments, like `VerifiedAccumulatorMessage` . I wouldn't worry about this for this PR though, since this change is already pretty big.

Reply:
you could potentially check some of the header bytes and data sources and stuff when that struct is created as well. That would consolidate the security checks into one place.

Reply:
yeah I think that could achieve better conceptual separation, in a later PR I could create a separate public entry-point function

`create_verified_accumulator_message`

and the user can create a programmable txn to call that and obtain a `VerifiedAccumulatorMessage` to be fed into this function